### ISSUE TITLE
Annotation callout closes automatically when changing the viewport

### ISSUE THREAD

**Issue by incanus** [mapbox/mapbox-gl-native#3154]
Android now does this. 

Complicated by the fact that `SMCalloutView` doesn't support a move API, just a placement/initial layout. On raster we used to use this repeatedly during viewport changes to sync up, but it became non-performant as of last check. 

Refs https://github.com/mapbox/mapbox-gl-native/issues/1125 https://github.com/mapbox/mapbox-gl-native/pull/2412

**Comment by 1ec5** [comment]
Also leave it open when the user location changes: #2618.

**Comment by silviokuehn** [comment]
I have the same problem. Auto-closing of the annotation callouts while after the map was moved (by app or manually by user interaction) seems to be a bad user experience. Moreover when you have a click event (button) within the callout bubble and user does a little bit move the finger while clicking, the pop-up disappears and the click event does not work.

As a workaround we tried a re-select of last selected annotation after a map move but that does not work really good.
